packages/svelte/src/compiler/phases/3-transform/client/visitors/javascript-runes.js

```diff
@@ -315,7 +315,16 @@ export const javascript_visitors_runes = {
 						)
 					);
 					for (let i = 0; i < bindings.length; i++) {
-						bindings[i].expression = b.member(b.call('$.get', b.id(id)), b.literal(i), true);
+						declarations.push(
+							b.declarator(
+								b.id(id + '_' + i),
```

Comment:
Can we use the original variables names instead? E.g.
```js
let { a, b } = $derived(..);
// becomes
let derived_value = $.derived(..);
let a = $.derived(() => $.get(derived_value)[0]);
let b = $.derived(() => $.get(derived_value)[1]);
```
Then we also might not need to set `expression` for them because assuming they are marked as being a derived value they code generation should add the `$.get(..)` automatically.

Reply:
Ah yes! Just applied :)